packages/repeater/src/request-runner/protocols/HttpRequestRunner.spec.ts

```diff
@@ -0,0 +1,187 @@
+import { HttpRequestRunner } from './HttpRequestRunner';
+import { Request, RequestOptions } from '../Request';
+import { RequestRunnerOptions } from '../RequestRunnerOptions';
+import { Protocol } from '../../models';
+import nock from 'nock';
+import 'reflect-metadata';
+import { anything, spy, verify, when } from 'ts-mockito';
+import { Logger, LogLevel } from '@secbox/core';
+import { SocksProxyAgent } from 'socks-proxy-agent';
+
+const createRequest = (options: Partial<RequestOptions> = {}) => {
+  const requestOptions = {
+    url: 'https://foo.bar',
+    method: 'GET',
+    headers: {},
+    ...options
+  };
+  const request = new Request(requestOptions);
+  const spiedRequest = spy(request);
+  when(spiedRequest.method).thenReturn('GET');
+
+  return { requestOptions, request, spiedRequest };
+};
+
+describe('HttpRequestRunner', () => {
+  const setupRunner = (
+    options: RequestRunnerOptions = {},
+    logger: Logger = new Logger()
+  ): HttpRequestRunner => new HttpRequestRunner(options, logger);
+
+  beforeAll(() => nock.disableNetConnect());
+  afterAll(() => nock.enableNetConnect());
+
+  beforeEach(() => nock.cleanAll());
```

Comment:
You have to remove an interceptor
```suggestion
  afterEach(() => {
    nock.cleanAll();
    nock.restore();
  });
  
  beforeEach(() => {
    if (!nock.isActive()) {
      nock.activate();
    }
  })
```